File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>map;
        int size = nums.size() , index = 0;

        for(int x : nums){
            int remain = target - x;

            if(map.find(remain) != map.end()){
                return {map[remain], index};
            }

            map[x] = index++; 
        }

        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        map = {}

        for index,value in enumerate(nums):
            remain = target - value
            if remain in map:
                return [map[remain] , index]
            map[value] = index
        
        return [-1,-1]